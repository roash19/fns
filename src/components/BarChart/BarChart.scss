@import "../../stylesheet/colors";
@import "../../stylesheet/mixin";

.bar-chart {
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
  border-bottom: 1px solid $black;
  border-left: 1px solid $black
}

.bar {
  display: flex;
  width: 50px;
  min-width: 25px;
  border: 1px solid $black;
  border-bottom: none;
  background-color: $blue;
  margin: 0 5px;
  writing-mode: vertical-rl;
  align-items: center;
  padding-top: 5px;

  &__text {
    @include font(15px, $white);
  }
}