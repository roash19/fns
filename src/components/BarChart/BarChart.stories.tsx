import React from 'react';
import { storiesOf } from '@storybook/react';
import { BarChart } from './BarChart';
import '../../stylesheet/index.scss';

storiesOf('BarChart', module).add('default', () => (
  <BarChart points={[10, 20, 30]}>Button</BarChart>
));
