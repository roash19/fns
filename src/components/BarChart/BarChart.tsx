import React from 'react';
import './BarChart.scss';
import cm from 'classnames';

interface BarChartProps {
  points: number[];
  className?: string;
}

export const BarChart: React.FC<BarChartProps> = ({ points, className }) => (
  <div className={cm(className, 'bar-chart')}>
    {points.map((point, i) => (
      // индекс в ключе, потому что статический список
      <div className="bar" key={i} style={{ height: `${point}%` }}>
        <p className="bar__text">{point}</p>
      </div>
    ))}
  </div>
);
