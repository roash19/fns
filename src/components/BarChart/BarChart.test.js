import React from 'react';
import renderer from 'react-test-renderer';
import { BarChart } from './BarChart';

it('renders correctly', () => {
  const tree = renderer
    .create(<BarChart points={[10, 20, 30]}>Facebook</BarChart>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});