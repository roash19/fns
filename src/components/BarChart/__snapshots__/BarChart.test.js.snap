// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bar-chart"
>
  <div
    className="bar"
    style={
      Object {
        "height": "10%",
      }
    }
  >
    <p
      className="bar__text"
    >
      10
    </p>
  </div>
  <div
    className="bar"
    style={
      Object {
        "height": "20%",
      }
    }
  >
    <p
      className="bar__text"
    >
      20
    </p>
  </div>
  <div
    className="bar"
    style={
      Object {
        "height": "30%",
      }
    }
  >
    <p
      className="bar__text"
    >
      30
    </p>
  </div>
</div>
`;
